### ISSUE TITLE
[bzlmod bug/regression] `entry_point` no longer working with blzmod 

### ISSUE THREAD

**Issue by xinbinhuang** [bazelbuild/rules_python#958]
# 🐞 bug report

### Affected Rule

 The issue is caused by the rule:  
`entry_point` when using with blzmod

### Is this a regression?

Yes

### Description

`entry_point` rule no longer working when used with blzmod

## 🔬 Minimal Reproduction

You can find a reproducible example at this repo: https://github.com/xinbinhuang/rules_python_bug

## 🔥 Exception or Error

```
~ $ bazel run "//:flake8"    
ERROR: /workspace/BUILD.bazel:3:6: no such package '@[unknown repo 'pip_flake8' requested from @]//': The repository '@[unknown repo 'pip_flake8' requested from @]' could not be resolved: No repository visible as '@pip_flake8' from main repository and referenced by '//:flake8'
ERROR: Analysis of target '//:flake8' failed; build aborted: Analysis failed
INFO: Elapsed time: 4.436s
INFO: 0 processes.
FAILED: Build did NOT complete successfully (1 packages loaded, 0 targets configured)
ERROR: Build failed. Not running target
```
## 🌍  Your Environment

**Operating System:**

<pre>
  <code>
mac/linux
  </code>
</pre>

**Output of `bazel version`:**

<pre>
  <code>
6.0.0
  </code>
</pre>

**Rules_python version:**

<pre>
  <code>
0.16.1
  </code>
</pre>

**Anything else relevant?**

**Comment by keith** [comment]
Same here, our case is even just registering `pre-commit` as a dep seems to get dropped

**Comment by keith** [comment]
my case was fixed at head with https://github.com/bazelbuild/rules_python/commit/2c29f401dcab6e6b84b56d9b4e9849fe0cf0a550, but based on that commit it's because my case had a `-` and yours doesn't, so maybe it's different. have you tried with HEAD?

**Comment by xinbinhuang** [comment]
> my case was fixed at head with [2c29f40](https://github.com/bazelbuild/rules_python/commit/2c29f401dcab6e6b84b56d9b4e9849fe0cf0a550), but based on that commit it's because my case had a `-` and yours doesn't, so maybe it's different. have you tried with HEAD?

Yeah, I've encountered that with another package and fixed it with the commit. But this's a different issue.

**Comment by chrislovecnm** [comment]
@aignas is this fixed?

**Comment by aignas** [comment]
Fixed by #1220.
